src/common.h

```diff
@@ -277,6 +527,40 @@ extern void get_system_HZ(void);
 #endif
 #endif
 
+/**
+ * Read `filename` containing a single number and return it.
+ *
+ * Open `filename`, read the first line, convert it to a number and store it in result.
+ *
+ * @param filename File to read.
+ * @param result Long to store the number in.
+ * @return 0 on success, -1 if could not open file, -2 if could not read from file.
+ */
 extern int read_single_number_file(const char *filename, unsigned long long *result);
 
+/// \todo ktsaou: your help needed
+typedef enum {
+    NETDATA_SIMPLE_PATTERN_MODE_EXACT,    ///< \todo ktsaou: your help needed
+    NETDATA_SIMPLE_PATTERN_MODE_PREFIX,   ///< \todo ktsaou: your help needed
+    NETDATA_SIMPLE_PATTERN_MODE_SUFFIX,   ///< \todo ktsaou: your help needed
+    NETDATA_SIMPLE_PATTERN_MODE_SUBSTRING ///< \todo ktsaou: your help needed
+} NETDATA_SIMPLE_PREFIX_MODE;
+typedef void NETDATA_SIMPLE_PATTERN; ///< \todo ktsaou: you help needed
```

Comment:
This is a way to prevent exposing the internals of simple patterns to the rest of netdata. So, all the calls accept `SIMPLE_PATTERN *`, which is actually a `void *`.